contracts/ConfidentialERC20/ConfidentialERC20.sol

```diff
@@ -109,8 +114,7 @@ abstract contract ConfidentialERC20 is Ownable, IConfidentialERC20, IERC20Metada
     function transfer(address to, euint64 value) public virtual returns (bool) {
         require(TFHE.isSenderAllowed(value));
         address owner = _msgSender();
-        ebool isTransferable = TFHE.le(value, _balances[msg.sender]);
```

Comment:
I did not like this essential, universal check being the responsibility of the caller, so it is moved to `_transfer` and the logic is centralised in `_hasSufficientBalance` which also considers the `_lockedBalances`